src/main/java/Util/Storage.java

```diff
@@ -0,0 +1,183 @@
+package Util;
+
+import Tasks.*;
+
+import java.io.*;
+
+import java.util.*;
+
+import java.util.Scanner;
+
+import Exception.DukeException;
+
+/**
+ * Storage class to handle all interactions between saving, reading and writing
+ * the file onto local disk. The interactions will be based on what is stored on
+ * TaskList
+ */
+public class Storage {
+
+    private Scanner scanner;
+    private PrintWriter pw;
+    private BufferedWriter bw;
+    private FileWriter fw;
+    private ArrayList<Task> storedList = new ArrayList<>();
+    private String filePath;
+
+    /**
+     * Constructs a Storage object to interact with given file
+     *
+     * @param filePath represents the file path/directory for which the
+     * writer/reader interacts with
+     */
+    public Storage(String filePath) throws Exception{
+        this.filePath = filePath;
+        File file = new File(filePath);
+        try {
+            if(!file.exists()) {
+                file.createNewFile();
+            }
+            scanner = new Scanner(new File(filePath));
+        } catch (Exception e) {
+            System.out.println("Error in reading file");
+        }
+
+        fw = new FileWriter(file, true);
+        bw = new BufferedWriter(fw);
+        pw = new PrintWriter(bw);
+    }
+
+    /**
+     * Reads all data from the file filePath
+     *
+     * @return ArrayList of tasks to be stored under TaskList
+     */
+    public ArrayList<Task> readFile() {
```

Comment:
You can avoid all these by making Task(s) Serializable, then you can use the API functions to dump (and read) the arraylist to disk. This has the benefit that if you add additional members to any Task(s), it will adapt automatically, although it will deprecate the old saves.